### PR TITLE
HTML-only molecule viewer

### PR DESCRIPTION
**Pull request by sallai**
viewer.visualize() is a function that turns the JSON string representation of a molecule into an HTML file, copies it, along with the requried css and js soruces to a temp folder, and opens it up in the browser. The JSON is embedded in the HTML so that we have no issues with the browsers' same-origin requirements.

### CODE REVIEW

**Comment by patrickfuller** [comment]
@sallai -Thanks for the pull request!

I'm curious to see if this could be wrapped into an existing function. Would you mind providing some information about your intended use case?

**Comment by sallai** [comment]
The tool we're building ( https://github.com/iModels/mbuild ) needs to visualize molecules not just from within ipython, but also from python programs running from the command line. So far, we've been using VMD, but popping up a browser window with imolecule in a static html allows me to ditch VMD from our project altogether.

Regarding wrapping this functionality into an existing function: yes, it actually does make sense to move it to notebook.draw(). The PR is coming.

**Comment by patrickfuller** [comment]
Looks good to me! I'm going to make a couple of nitpicky comments, but I'm happy to merge after that.

File: imolecule/notebook.py [hunk 1]
```diff
@@ -88,7 +88,41 @@ def draw(data, format="auto", size=(400, 300), drawing_type="ball and stick",
...
+            # we're running outside ipython, let's generate a static HTML and show it in the browser
+            import shutil
+            import webbrowser
+            import tempfile
+            import time
+            import urlparse
+            import urllib
+            from tornado import template
+
+            PATH = os.path.dirname(os.path.realpath(__file__))
```
Comment (patrickfuller) [inline imolecule/notebook.py]: I think the `file_path` variable works as a replacement here.

File: imolecule/notebook.py [hunk 2]
```diff
@@ -88,7 +88,41 @@ def draw(data, format="auto", size=(400, 300), drawing_type="ball and stick",
...
+            html = t.generate(title="imolecule", json_mol=json_mol, drawing_type=drawing_type,
+                           camera_type=camera_type, shader=shader)
+
+            tempdir = tempfile.mkdtemp(prefix='imolecule_' + str(int(time.time())) + '_')
+
+            html_filename = os.path.join(tempdir,'index.html')
+            with open(html_filename, 'w') as f:
+                f.write(html)
+
+            shutil.copy(os.path.join(PATH, os.path.join('server/css', 'chosen.css')), tempdir)
```
Comment (patrickfuller) [inline imolecule/notebook.py]: You can chain `os.path.join` into something like `os.path.join(file_path, 'server', 'css', 'chosen.css')`

**Comment by sallai** [comment]
Great! Comments have been addressed.

**Comment by patrickfuller** [comment]
I just realized that I was ambiguous on "I think the file_path variable works as a replacement here." comment. There's a `file_path` variable defined [here](https://github.com/patrickfuller/imolecule/blob/master/imolecule/notebook.py#L10) that you could use without having to redefine it.

I want to take some code cleaning passes on this project anyway, so I'll update when I get around to that.

**Pull request merged**